### ISSUE TITLE
[feature] Code upload instead of docker push workflow

### ISSUE THREAD

**Issue by marrrcin** [getindata/kedro-vertexai#82]
This idea is borrowed from Azure ML (and this PR https://github.com/getindata/kedro-azureml/pull/15 ) - where you define an `Environment`, which is a docker image which runs your image, but the code is not part of the image (only dependencies are present in the image).
The workflow for that will make Data Science iterations faster, as they will not have to build the docker image every time they want to run / debug something in Vertex AI. This issue itself will be partially addressed by #81 , but this would be a next iteration on that.

General workflow would work like this:
1. Docker image with dependencies is uploaded to the container registry.
1. User runs `kedro vertexai run-once` with some flag (or maybe we should have `kedro vertexai run` for docker and `kedro vertexai run-once` for this flow 💡)
2. The code of Kedro project is copied to GCS (first packaged and compressed) and the job is started within the container. The container should have a modified entrypoint which will first download the code from GCS and then execute it.

Please discuss the design with @em-pe  and @szczeles before implemeting.

**Comment by adrienpl** [comment]
This feature is really a must have to have a functional development environment with kedro.

**Comment by szczeles** [comment]
@adrienpl Agree! The current development cycle with docker images is so painful...

Actually, I've been working on some implementation of this feature a year ago. Once I find the local branch, I will push it so somebody can take it over (I'm not into kedro/vertex anymore).

**Comment by Dahll** [comment]
Thank you ! It will be really helpfull. What are you using right now ?

**Comment by adrienpl** [comment]
Thank you ! It will be really helpfull. What are you using right now ?

**Comment by szczeles** [comment]
@adrienpl I'm no longer on a project in MLOps area, so I'm using just boring dev tools ATM ;-)

**Comment by em-pe** [comment]
Seems like a stale issue, comment if you think we should reopen.
